pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -156,85 +146,191 @@ def remove_module_from_repo(self, repo, criteria):
         return self.client.get(repo['_href'], params={'details': True})
 
 
-class CopyModulesTestCase(unittest.TestCase):
-    """Test copy of modules, and its artifacts.
+class CopyModularDefaultsTestCase(unittest.TestCase):
+    """Test ``recursive`` and ``recursive_conservative`` flags during copy.
 
-    This test targets the following issue:
+    This test targets the following issues:
 
-    `Pulp Smash #1122 <https://github.com/PulpQE/pulp-smash/issues/1122>`_
+    * `Pulp Smash #1122 <https://github.com/PulpQE/pulp-smash/issues/1122>`_
+    * `Pulp #4543 <https://pulp.plan.io/issues/4543>`_
+
+    Recursive or conservative copy of ``modulemd_defaults``
+    should always only copy:
+
+    * 3 modules: ``duck``, ``kangaroo``, ``walrus``
+
+    Exercise the use of ``recursive and ``recursive_conservative``.
     """
 
     @classmethod
     def setUpClass(cls):
         """Create class wide variables."""
         cls.cfg = config.get_config()
-        if cls.cfg.pulp_version < Version('2.17'):
-            raise unittest.SkipTest('This test requires Pulp 2.17 or newer.')
+        if cls.cfg.pulp_version < Version('2.19'):
+            raise unittest.SkipTest('This test requires Pulp 2.19 or newer.')
+        if check_issue_4405(cls.cfg):
+            raise unittest.SkipTest('https://pulp.plan.io/issues/4405')
         cls.client = api.Client(cls.cfg, api.json_handler)
-
-    def test_copy_modulemd_recur(self):
-        """Test copy of modulemd in RPM repository in recursive mode."""
-        criteria = {
-            'filters': {'unit': {
-                'name': MODULE_FIXTURES_PACKAGE_STREAM['name'],
-                'stream': MODULE_FIXTURES_PACKAGE_STREAM['stream']
-            }},
-            'type_ids': ['modulemd'],
+        cls.criteria = {
```

Comment:
Is there a special reason to be a class variable? Could we move this variable to the ``copy_units``, since they use the same criteria?